apps/ecommerce/lambda/src/controllers/apiController.ts

```diff
@@ -1,49 +1,54 @@
-import { Request, Response } from 'express';
-import axios, { AxiosResponse } from 'axios';
-import { HydratedResourceData, ResourceLink } from '@/src/types';
+import { NextFunction, Request, Response } from 'express';
+import axios, { AxiosError, AxiosResponse } from 'axios';
+import { HydratedResourceData, ExternalResourceLink } from '@/src/types';
 import { config } from '../config';
-const BASE_URL = config.baseUrl;
+const BASE_URL = `${config.baseUrl}${config.stage === 'prod' ? '' : `/${config.stage}`}`;
 
-interface Provider {
-  name: string;
-  resourceURL: string;
-}
-
-const PROVIDERS = [
-  {
-    name: 'shopify',
-    resourceURL: `${BASE_URL}/shopify/resource`,
-  },
-];
+const PROVIDERS: Record<string, string> = {
+  shopify: `${BASE_URL}/shopify/resource`,
+};
 
 const ApiController = {
   ping: (req: Request, res: Response) => {
     return res.send({ status: 'ok', message: 'pong' });
   },
+
   resource: async (
     req: Request,
-    res: Response<HydratedResourceData | { error?: unknown; message: string }>
+    res: Response<HydratedResourceData | { error?: unknown; message: string }>,
+    next: NextFunction
   ) => {
-    const body: ResourceLink = req.body;
-    const provider = body.sys.provider;
-    const entityType = body.sys.linkType;
-    const index = PROVIDERS.findIndex((p: Provider) => p.name === provider?.toLowerCase());
+    try {
+      const resourceLink: ExternalResourceLink = req.body;
+      const proxyUrl = PROVIDERS[resourceLink.sys.provider.toLowerCase()];
+
+      if (!proxyUrl) {
+        return res.status(404).send({
+          status: 'error',
+          message: `Provider${
+            resourceLink.sys.provider ? `: ${resourceLink.sys.provider}` : ''
+          } not found`,
+        });
+      }
 
-    if (index > -1) {
       try {
-        const url = PROVIDERS[index].resourceURL;
-        const response: AxiosResponse<HydratedResourceData> = await axios.post(url, body);
-        res.status(response.status).send(response.data);
+        let response;
+        try {
+          response = await axios.post(proxyUrl, resourceLink);
+        } catch (error) {
+          response = (error as AxiosError).response;
```

Comment:
Axios throws an error object that contains the original response in `error.response` – since we want this to operate a passthrough proxy call, we have to explicitly catch and set our response to this here. cc: @pkeavenycontentful